### DISCUSSION TITLE
ACME protocol support

### DISCUSSION THREAD

**Discussion by malthe** [bfenetworks/ingress-bfe#65] [Ideas]
Adding built-in ACME protocol support would alleviate the need for a separate process to keep certificates updated.

References:

- https://github.com/caddyserver/ingress#automatic-https
- https://letsencrypt.org/docs/challenge-types/

**Comment by daimg** [comment]
Thanks for your attention on this project, we will add it into TODO list. 
Because some tool, such as [cert-manager](https://cert-manager.io/), could perform such functionality automatically, it would have low priority.

**Reply by malthe** [reply]
@daimg note that the [TLS-ALPN-01](https://letsencrypt.org/docs/challenge-types/#tls-alpn-01) challenge must be supported directly by the ingress controller because it works directly on the TLS layer.

**Reply by daimg** [reply]
yes, cert-manager doesn't support TLS-ALPN-01 yet, while challenges DNS-01 and HTTP-01 have to be used. 
TLS-ALPN-01 requires more special logic in TLS handshake for this new APLN defined in spec, we may need some enhancement in [BFE](https://github.com/bfenetworks/bfe) eventually,  which actually process TLS handshake.
